docs/DeveloperGuide.md

```diff
@@ -158,10 +158,52 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### \[Proposed\] Grouping Feature
 
 #### Proposed Implementation
 
+The proposed grouping mechanism is facilitated by `GroupedUniquePersonList`. It extends `UniquePersonList` with task that are linked between the people of the same group. Additionally, it implements the following operations:
+
+* `GroupedUniquePersonList#assignTask(Task)` - Add task to everyone in the group
+* `GroupedAddressBook#markTask(Index)` - Mark task of everyone in the group
+* `GroupedAddressBook#unmarkTask(Index)` - Unmark task of everyone in the group
+
+These operations are exposed in the `Model` interface as `Model#assignTaskToGroup(String, Task)`, `Model#markTaskOfGroup(String, Index)` and `Model#unmarkTaskofGroup(String, Index)` respectively.
+
+`GroupedUniquePersonList` adds a new string called `groupName` to label each of their groups.
+
+A new list of `GroupedUniquePersonList` will be added to the `Model` interface.
+
+To add to the list of `GroupedUniquePersonList`, the Model interface includes `Model#addGroup(String, List<Person>)` and `Model#addListOfGroups(List<Group>)`.
+
+To remove to the list of `GroupedUniquePersonList`, the Model interface includes `Model#removeGroup(String)`.
+
+New operation are exposed in the `Model` interface are `Model#addPersonToGroup(String, Person)`, `Model#removePersonFromGroup(String, Person)` and `Model#deleteAssignedTaskGroup(String, Task)` which would call `UniquePersonList#add(Person)`, `UniquePersonList#remove(Person)` and `UniquePersonList#deleteAssignedTask(Person)`respectively.
+
+Given below is an example usage scenario and how the grouping mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The list of the `GroupedUniquePersonList` will be empty if there are no groups stored in the storage.
+
+<puml src="diagrams/GroupingState0.puml" alt="GroupingState0" />
+
+Step 2. The user executes `group gn/2103T gp/Ivan gp/Greg gp/Dave` command to group Ivan, Greg and Dave from the displayed person list to one group. The `group` command calls `Model#addGroup(String, List<Person>)`, which creates a new group with that contains the list of people that was indicated by the user.
```

Comment:
Just a minor typo: `a new group with that contains the list` -> `a new group that contains the list`.